lib/swagger-ui/swagger-ui.ts

```diff
@@ -0,0 +1,72 @@
+import * as swaggerUi from 'swagger-ui-dist';
+import { favIconHtml, htmlTemplateString, jsTemplateString } from './constants';
+import { SwaggerCustomOptions } from '../interfaces';
+
+const buildJSInitOptions = (obj) => {
+  const placeholder = '____FUNCTIONPLACEHOLDER____';
+  const fns = [];
+  let json = JSON.stringify(
+    obj,
+    function (key, value) {
+      if (typeof value === 'function') {
+        fns.push(value);
+        return placeholder;
+      }
+      return value;
+    },
+    2
+  );
+
+  json = json.replace(new RegExp('"' + placeholder + '"', 'g'), function (_) {
+    return fns.shift();
+  });
+
+  return `let options = ${json};`;
+};
+
+export const buildSwaggerHTML = function (
+  baseUrl,
+  swaggerDoc,
+  options: SwaggerCustomOptions
+) {
+  const customCss = options?.customCss ?? '';
+  const customJs = options?.customJs ?? '';
+  const customfavIcon = options?.customfavIcon ?? false;
+  const customSiteTitle = options?.customSiteTitle ?? 'Swagger UI';
+  const customCssUrl = options?.customCssUrl ?? '';
```

Comment:
We can simplify thise whole block to:
```suggestion
  const { customCss, customJs, customfavIcon, customSiteTitle, customCssUrl } = options;
```

And them assign the defaults where you're using them, with the `??` operator